Classes/Repository/RepositoryCodeDirectoryViewController.swift

```diff
@@ -130,24 +130,53 @@ ListSingleSectionControllerDelegate {
 
     func didSelect(_ sectionController: ListSingleSectionController, with object: Any) {
         guard let file = object as? RepositoryFile else { return }
-        let next = path.appending(file.name)
-        let controller: UIViewController
+        let nextPath = path.appending(file.name)
+
         if file.isDirectory {
-            controller = RepositoryCodeDirectoryViewController(
-                client: client,
-                repo: repo,
-                branch: branch,
-                path: next
+            showDirectory(at: nextPath)
+        } else {
+            showFile(at: nextPath)
+        }
+    }
+
+}
+
+// MARK: - RepositoryCodeDirectoryViewController (Navigation) -
+
+extension RepositoryCodeDirectoryViewController {
+
+    private func showDirectory(at path: FilePath) {
+        let controller = RepositoryCodeDirectoryViewController(
+            client: client,
+            repo: repo,
+            branch: branch,
+            path: path
+        )
+
+        navigationController?.pushViewController(controller, animated: trueUnlessReduceMotionEnabled)
+    }
+
+    private func showFile(at path: FilePath) {
```

Comment:
Maybe we can unify these `if-else` paths a bit more by setting just the `ViewController` to present:

```swift
let viewController: UIViewController
if path.hasBinarySuffix {
    viewController = RepositoryWebViewController(
                                     repo: repo,
                                     branch: branch,
                                     path: path
                                 )
} else {
    viewController = RepositoryCodeBlobViewController(
                                     client: client,
                                     repo: repo,
                                     branch: branch,
                                     path: path
                                 )
}
navigationController?.pushViewController(viewController, animated: trueUnlessReduceMotionEnabled)
```